sklearn/ensemble/gradient_boosting.py

```diff
@@ -965,8 +966,41 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # fit initial model - FIXME make sample_weight optional
             self.init_.fit(X, y, sample_weight)
 
-            # init predictions
-            y_pred = self.init_.predict(X)
+            if is_classifier(self.init_): 
+                n_classes = np.unique(y).shape[0]
+            else:
+                n_classes = 1
+
+            # If the initialization estimator has a predict_proba method,
+            # either use those, or collapse to a single vector if there
+            # are only two classes
+            if hasattr(self.init_, 'predict_proba'):
+                eps = np.finfo(X.dtype).eps
+                y_pred = self.init_.predict_proba(X) + eps
+                if n_classes == 2:
+                    y_pred = np.log(y_pred[:,1] / y_pred[:,0])
+                    y_pred = y_pred.reshape(n_samples, 1)
+
+            # Otherwise, it can be a naive estimator defined above, in which
+            # case don't do anything, or a classifier whose estimates will be
+            # a vector that should be hot encoded, or a regressor whose
+            # estimates still need to be reshaped from (n_samples,) to
+            # (n_samples,1)
+            else:
+                pred = self.init_.predict(X)
+
+                if len(pred.shape) < 2:
+                    if is_classifier(self.init_):
+                        y_pred = np.zeros((n_samples, n_classes))
+                        y_pred[:, pred] = 1.0
+                        if n_classes == 2:
+                            y_pred = np.log(y_pred[:,1] / y_pred[:,0])
```

Comment:
I suppose smoothing by an epsilon. This case is not covered in the unit tests; maybe it should be added as well.